### DISCUSSION TITLE
display the preset number of the pin mapping on the OLED screen

### DISCUSSION THREAD

**Discussion by chaso14** [OpenStickCommunity/GP2040-CE#801] [Q&A]
I apologize if this has been covered before. I couldn't find anything relevant when I searched the forum for display-related topics. I'm Japanese and using a translation tool to post this, so please feel free to ask if anything is unclear.

My OLED display already shows the default button layout and connection method. However, I use different pin mapping profiles for different situations: Profile 1 for Xinput on PC, and Profile 2 (with limited buttons due to hardware constraints) for Dinput in Japanese game centers. I switch between these profiles using a hotkey.

I'd like to display the preset number of the pin mapping on the OLED, so I can easily see which profile is active. This would be really convenient for me. Is there a relatively simple way to implement this?

If anyone knows of a method to achieve this, or if there's already a solution out there, please let me know.

Thank you for your assistance!

**Comment by bsstephan** [comment]
@chaso14 - please make a feature request under the GP2040-CE project's Issues. This request has been made a couple times, but I'm not sure what the state of it is anymore.

**Comment by TheTrainGoes** [comment]
Hi @chaso14 ,

To be clear, when you say `preset number of the pin mapping` you are looking to have the profile number displayed on the screen correct?

**Reply by chaso14** [reply]
Yes, I would like to check which mode of Pin Mapping Profile1~4 it is in.
It would be even better if you could give your profile a name.

**Comment by TheTrainGoes** [comment]
This has all been implemented into the firmware - closing
